dagshub/data_engine/model/query_result.py

```diff
@@ -401,6 +404,78 @@ def download_binary_columns(
             num_proc=num_proc,
         )
 
+    def predict_with_mlflow_model(
+        self,
+        repo: str,
+        name: str,
+        version: str = "latest",
+        pre_hook: Callable = lambda x: x,
+        post_hook: Callable = lambda x: x,
+        batch_size: int = 1,
+        log_to_field: str = None,
+        return_predictions: bool = True,
+    ) -> Optional[list]:
+        """
+        Sends all the datapoints returned in this QueryResult as prediction targets for
+        an MLFlow model registered on DagsHub.
+
+        Args:
+            repo: repository to extract the model from
+            name: name of the model in the mlflow registry
+            version: (optional, default: 'latest') version of the model in the mlflow registry
+            pre_hook: (optional, default: identity function) function that runs before datapoint is sent to the model
+            post_hook: (optional, default: identity function) function that converts mlflow model output
+            to the desired format
+            batch_size: (optional, default: 1) function that sets batch_size
+            log_to_field: (optional, default: 'prediction') write prediction results to metadata logged in data engine.
+            If None, just returns predictions.
+            return_predictions: (optional, default: True) returns prediction result
+            (in addition to logging to a field, iff that parameter is set)
+        """
+        if not log_to_field and not return_predictions:
+            raise ValueError("Either `log_to_field` or `return_predictions` must be set.")
+
+        # to support depedency-free dataloading, `Batcher` is a barebones dataloader that sets up batched inference
+        class Batcher:
+            def __init__(self, dset, batch_size):
+                self.dset = dset
+                self.batch_size = batch_size
+
+            def __iter__(self):
+                self.curr_idx = 0
+                return self
+
+            def __next__(self):
+                self.curr_idx += self.batch_size
+                return [self.dset[idx] for idx in range(self.curr_idx - self.batch_size, self.curr_idx)]
+
+        init(*repo.split("/")[::-1])
+        model = mlflow.pyfunc.load_model(f"models:/{name}/{version}")
+        init(*self.datasource.repo.split("/")[::-1])
+
```

Comment:
Side effects of this:
1) Visual noise with the "Initialized repo blabla" from every init
2) If the user hasn't been using mlflow to track stuff before on this repo, now they do

Did you try going the MlflowClient route?
I saw that you tried to use `set_tracking_uri`, and honestly I think it'll be less noisy to use that function instead of init, if MlflowClient can't be used/too hard to use